### DISCUSSION TITLE
Magic comment for Versicle lets the app hanging

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#3602] [Q&A]
Hi everyone, 
in my attempts to incorporate more and more of the Cistercian Rite into DivinumOfficium.com, I have hit a strange hurdle:
the Versicles in Lauds and Vespers are quite often different from the Roman version. In various Feasts and Commons, the remedy is quite simple, however for the Ferial Office, it get's complicated. In the past, I changed the code for Versicles of the main Office and Commemorations and added the `VersumC` label for the Cistercian version in `Major Special.txt`, which did the trick quite fine. But then it hit me that I could use magic comment `(rubrica cisterciensis)`, which would make the code cleaner and it would also eliminate the need for separate VersumC, where it just copies the Roman Office... 

So I tried it, but instead of it just working, the Perl interpreter just hangs. What can I do to make it right? This seems like some counter trying to hit infinity, until the system kills it... I will get turn the code back now, but I'd like to have this working, however I don't really know, where the magic comments are implemented, so it seems I need help.

Thanks for replies, have a nice day!

**Comment by FAJ-Munich** [comment]
In `dialogcommon.pl` there is this passage:
```
our %predicates = (
  tridentina => sub { shift =~ /Trident/ },
  monastica => sub { shift =~ /Monastic/ },
  innovata => sub { shift =~ /NewCal/i },
  innovatis => sub { shift =~ /NewCal/i },
  paschali => sub { shift =~ /Paschæ|Ascensionis|Octava Pentecostes/i },
  'post septuagesimam' => sub { shift =~ /Septua|Quadra|Passio/i },
  prima => sub { shift == 1 },
  secunda => sub { shift == 2 },
  tertia => sub { shift == 3 },
  longior => sub { shift == 1 },
  brevior => sub { shift == 2 },
  'summorum pontificum' => sub { ${shift()}{summpont} },
);
```

**Reply by Augustinus-Altovadensis** [reply]
Yes, and my %predicates look like this:
```
our %predicates = (
  tridentina => sub { shift =~ /Trident/ },
  cisterciensis => sub { shift =~ /Cistercien/i },
  monastica => sub { shift =~ /Monastic/ },
  innovata => sub { shift =~ /NewCal/i },
  innovatis => sub { shift =~ /NewCal/i },
  paschali => sub { shift =~ /Paschæ|Ascensionis|Octava Pentecostes/i },
  'post septuagesimam' => sub { shift =~ /Septua|Quadra|Passio/i },
  prima => sub { shift == 1 },
  secunda => sub { shift == 2 },
  tertia => sub { shift == 3 },
  longior => sub { shift == 1 },
  brevior => sub { shift == 2 },
  'summorum pontificum' => sub { ${shift()}{summpont} },
);
```

The strange thing is that it works most of the times, but for the Versus, it does work just sometimes - I got it working for Advent verses, but not for the Lent and Paschal ones. And also, if it doesn't work, it doesn't just output something or an error, but it overloads the `perl` process, and outputs just the first title line, nothing else. Strange.
